extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -864,6 +861,19 @@ export async function newDeployment(
     return undefined;
   }
 
+  try {
+    // Attempt to add the Config file to the files for deployment
+    // If the configuration is invalid, for example 'unknown', this will fail
+    await api.files.updateFileList(
+      configName,
+      getRelPathForConfig(configCreateResponse.configurationPath),
+      FileAction.INCLUDE,
+      newDeploymentData.entrypoint.inspectionResult.projectDir,
+    );
+  } catch (error: unknown) {
+    // continue on, as this is not a critical failure
+  }
```

Comment:
That is totally fair. I think in this case because these operations aren't completely blocking on the rest of the config + deployment file creation any error we get, even if it is unexpected, shouldn't stop the rest of the function from continuing.

My follow-up questions are two fold:
- do you agree with the above?
- what do you propose we add to the catch?

The most obvious addition could be some logging, but I didn't think showing a notification to the user or something that heavy was necessary for a non-blocking action failing.

Reply:
Yes, logging might be the best option here. Maybe also some more details in code comments about why this isn't and shouldn't be a blocking operation, this to help future collabs to better determine if this needs to change in case something else lands within the `try` block.

Reply:
Included some debug logging in https://github.com/posit-dev/publisher/pull/2422/commits/5241816b03e5cc9129607bb783ec313f15e87b8d

I also improved the comment a bit to be more explicit about why it is fine to continue on in the new deployment flow.